### ISSUE TITLE
Session cookie expires directive uses invalid format

### ISSUE THREAD

**Issue by MohsenRahimi** [spring-projects/spring-session#1514]
I recently upgrade one of our projects to use latest version 2.1.8 from 2.0.7 and we faced following issue:

in previous version it used to format the expiry date as GMT without time offset regardless of the server timezone. but last version will format the expiry in time offset representation (+HHMM) if the server time zone  is not GMT.

It's like Browsers (IE, Chrome) they don't like to see anything beside GMT. in our case the are expiring the cookie unexpectedly. 

for workaround we changed server time zone to GMT.

basically following line is responsible to format the time:

sb.append("; Expires=").append(expires.format(DateTimeFormatter.RFC_1123_DATE_TIME));

and browsers suppose to support RFC_1123 I guess, but it seems they don't like anything but GMT.

**Comment by vpavic** [comment]
Could you provide concrete cookies exactly how they appear for you in Spring Session 2.0 vs 2.1? Can you also provide some example that reproduces this problem?

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by vpavic** [comment]
I did some digging around this, and it seems that there is some difference in how we write session cookie expiration in 2.1 vs before:

This is what session cookie looks like in 2.0:

```
SESSION=NzkxYmFhYTYtY2ZiMC00N2UxLThlMDAtZGMzZmQ4M2FkZjdi; Max-Age=2147483647; Expires=Fri, 24-Oct-2087 18:50:40 GMT; Path=/; HttpOnly
2087-10-24T18:50:40.258Z
```

And this is 2.1:

```
SESSION=OGNhNDcxYTYtOTU0Yi00MzljLWIxMzctZjVmODcxMmEyNDc0; Max-Age=2147483647; Expires=Fri, 24 Oct 2087 20:55:19 +0200; Path=/; HttpOnly; SameSite=Lax
2087-10-24T18:55:19.064Z
```

I'm going to take a closer look at this. @MohsenRahimi, could you provide more info on how to reproduce this as that would be of real help here?

**Comment by vpavic** [comment]
After looking into it, this seems as a bug as `Expires` directive should be set according to:

- https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date

**Comment by vpavic** [comment]
Thanks for the report @MohsenRahimi, this is now fixed in `master` and `2.1.x`.
